allennlp/interpret/influence_interpreters/influence_interpreter.py

```diff
@@ -0,0 +1,111 @@
+from typing import List, Optional
+
+import torch
+
+from allennlp.common import Registrable
+from allennlp.common.util import JsonDict
+from allennlp.models.model import Model
+from allennlp.predictors import Predictor
+from allennlp.data.dataset_readers import DatasetReader
+from allennlp.data.data_loaders import MultiProcessDataLoader
+
+
+class InfluenceInterpreter(Registrable):
+    """
+    A `SaliencyInterpreter` interprets an AllenNLP Predictor's outputs by assigning an influence
+    score to each training instance with respect to each input.
+    """
+
+    def __init__(
+        self,
+        predictor: Predictor,
+        train_filepath: str,
+        test_filepath: str,
+        train_dataset_reader: DatasetReader,
+        test_dataset_reader: Optional[DatasetReader] = None,
+        params_to_freeze: List[str] = None,
+        k: int = 20,
+        device: int = -1,
+    ) -> None:
+        if type(device) is not int:
+            raise ValueError("'device' has to be int, -1 for cpu")
+        if type(device) is not int:
+            raise ValueError("'k' (i.e. number of supporting examples) has to be int")
```

Comment:
The canonical way to do this would be

```python
        if not isinstance(device, int):
            raise ValueError("'device' has to be int, -1 for cpu")
        if not isinstance(k, int):
            raise ValueError("'k' (i.e. number of supporting examples) has to be int")
```

That said, I don't think having these checks are necessary since we have type hints.

```suggestion
```